pkg/util/ecs/metadata/testutil/dummy_ecs.go

```diff
@@ -11,14 +11,20 @@ import (
 	"net/http"
 	"net/http/httptest"
 	"os"
+	"sync"
+	"time"
 )
 
 // DummyECS allows tests to mock ECS metadata server responses
 type DummyECS struct {
-	mux          *http.ServeMux
-	fileHandlers map[string]string
-	rawHandlers  map[string]string
-	Requests     chan *http.Request
+	sync.Mutex
+	mux               *http.ServeMux
+	fileHandlers      map[string]string
+	fileHandlersDelay map[string]time.Duration
+	rawHandlers       map[string]string
+	rawHandlersDelay  map[string]time.Duration
+	Requests          chan *http.Request
+	RequestCount      int
```

Comment:
I’m wondering if it would be interesting to use an `atomic` value here to avoid having to lock the mutex when incrementing this counter below [line 104](https://github.com/DataDog/datadog-agent/pull/28130/files#diff-96dd80736c4f64f87e639056330ba30a92a34dd3a530d4ad90f7bd238656cdf8R103-R105).

```suggestion
	RequestCount      atomic.Uint64
```